test/unit/core/selectors.ts

```diff
@@ -1,8 +1,29 @@
+import { MAX_RECORDS } from '../../../src/core/adapters/search';
 import Selectors from '../../../src/core/selectors';
 import suite from '../_suite';
 
 suite('selectors', ({ expect, stub }) => {
 
+  describe('searchRequest()', () => {
+    it('should decrease page size to prevent exceeding MAX_RECORDS', () => {
+      const originalPageSize = MAX_RECORDS - 1;
+
+      stub(Selectors, 'fields');
+      stub(Selectors, 'query');
+      stub(Selectors, 'collection');
+      stub(Selectors, 'selectedRefinements');
+      stub(Selectors, 'sort');
+      stub(Selectors, 'requestSort');
+      stub(Selectors, 'pageSize').returns(originalPageSize);
```

Comment:
I would say having both of these return the same value, even if it conveniently showcases the behaviour, makes it confusing for future developers to read this as it implies that `pageSize` and `skip` somehow reference the same data. Equally it's unclear the the `pageSize` is `1` because we have removed `1` from `MAX_RECORDS`